build.sbt

```diff
@@ -196,6 +196,8 @@ lazy val cli = crossProject(JVMPlatform, NativePlatform)
     ),
     scalacOptions ++= scalacJvmOptions.value,
     Compile / mainClass := Some("org.scalafmt.cli.Cli"),
+  ).nativeSettings(
+    nativeConfig ~= { _.withMode(Mode.releaseFull) },
     nativeImageInstalled := isCI,
     nativeImageOptions ++= {
```

Comment:
Similar issue as pointed out above, nativeImage settings will be ignored here (basically Native Image still uses classfiles to make a binary file, so it has to use the JVM target, Scala Native compiler outputs something else with the `.nir` extension, which then the linker uses that to make a binary file.